sender/nats.go

```diff
@@ -0,0 +1,171 @@
+	TLSCACert	string   `doc:"CA cert file path"`
+	UserCreds       string   `doc:"Nats credentials file path"`
+	NKeyFile        string   `doc:"Nats nkey file path"`
+	Insecure	bool	 `doc:"TLS InsecureSkipVerify"`
+	Encoder		skogul.EncoderRef
+	o		*[]nats.Option
+	nc		*nats.Conn
+	once		sync.Once
+}
+
+// Verify configuration
+func (n *Nats) Verify() error {
+        if n.Subject == "" {
+                return skogul.MissingArgument("Subject")
+        }
+
+        return nil
+}
+
+
+func (n *Nats) init() {
+
+	if n.Encoder.Name == "" {
+		n.Encoder.E = encoder.JSON{}
+	}
+
+	if n.Name == "" {
+		n.Name = "skogul"
+	}
+	n.o = &[]nats.Option{nats.Name(n.Name)}
+
+	if n.Servers == "" {
+		n.Servers = nats.DefaultURL
+	}
+
+	//User Credentials, use either.
+	if n.UserCreds != "" && n.NKeyFile != "" {
+		natsLog.Fatal("Please configure usercreds or nkeyfile.")
+	}
+	if n.UserCreds != "" {
+		*n.o = append(*n.o, nats.UserCredentials(n.UserCreds))
+	}
+
+	//Plain text passwords
+	if n.Username != "" && n.Password != "" {
+		if n.TLSClientKey != "" {
+			natsLog.Warnf("Using plain text password over a non encrypted transport!")
+		}
+		*n.o = append(*n.o, nats.UserInfo(n.Username, n.Password))
+	}
+
+	//TLS authentication, Note: Fix selfsigned certificates.
+	if n.TLSClientKey != "" && n.TLSClientCert != "" {
+		cert, err := tls.LoadX509KeyPair(n.TLSClientCert, n.TLSClientKey)
+		if err != nil {
+			natsLog.Fatalf("error parsing X509 certificate/key pair: %v", err)
+			return
+		}
+
+		cp, err := skogul.GetCertPool(n.TLSCACert)
+                if err != nil {
+                        natsLog.Fatalf("Failed to initialize root CA pool")
+			return
+                }
+
+		config := &tls.Config{
+			InsecureSkipVerify:	n.Insecure,
+			Certificates:		[]tls.Certificate{cert},
+			RootCAs:		cp,
+		}
+		*n.o = append(*n.o, nats.Secure(config))
+	}
+
+	//NKey auth
+	if n.NKeyFile != "" {
+		opt, err := nats.NkeyOptionFromSeed(n.NKeyFile)
+		if err != nil {
+			natsLog.Fatal(err)
+		}
+		*n.o = append(*n.o, opt)
+	}
+
+	//Log disconnects
+	*n.o = append(*n.o, nats.DisconnectErrHandler(func(nc *nats.Conn, err error) {
+                natsLog.WithError(err).Error("Got disconnected!")
+        }))
+	//Log reconnects
+	*n.o = append(*n.o, nats.ReconnectHandler(func(nc *nats.Conn) {
+                natsLog.Info("Reconnected")
+        }))
+	//Always try reconnecting
+	*n.o = append(*n.o, nats.RetryOnFailedConnect(true))
+	//Keep doing reconnects
+	*n.o = append(*n.o, nats.MaxReconnects(-1))
+
+	var err error
+	n.nc, err = nats.Connect(n.Servers, *n.o...)
+	if err != nil {
+		natsLog.Errorf("Encountered an error while connecting to Nats: %v", err)
+	}
+
+}
+
+
+
+func (n *Nats) Send(c *skogul.Container) error {
+	n.once.Do(func() {
+		n.init()
+	})
+	nm := make([]byte, 0, len(c.Metrics))
+	for _, m := range c.Metrics {
+		b, err := n.Encoder.E.EncodeMetric(m)
+		if err != nil {
+			return fmt.Errorf("couldn't encode metric: %w", err)
+		}
+		nm = append(nm, b...)
+	}
```

Comment:
I think you want to use just regular n.Encoder.E.Encode(c) for this. We use EncodeMetric(m) in kafka because Kafka has implicit batching, but here you are just concatinating each metric after the other, which means there's no clear separation between them and will probably make the parsing somewhat problematic.